src/qml/qmlconfigproxy.h

```diff
@@ -18,7 +21,32 @@ class QmlConfigProxy : public QObject {
     Q_INVOKABLE QVariantList getHotcueColorPalette();
     Q_INVOKABLE QVariantList getTrackColorPalette();
 
+    static QmlConfigProxy* create(QQmlEngine* pQmlEngine, QJSEngine* pJsEngine) {
+        Q_UNUSED(pQmlEngine);
+
+        // The instance has to exist before it is used. We cannot replace it.
+        Q_ASSERT(s_pInstance);
```

Comment:
> Why did you use Q_ASSERT instead of the custom DEBUG_ASSERT?

It's the code taken from last example in https://doc.qt.io/qt-6/qqmlengine.html#QML_SINGLETON .

> The definition of this function should be moved from the header into the .cpp file.

I'd like to deduplicate the code anyway. I tried with a template class and it compiled, but didn't work properly. Do you have an idea?

Reply:
No idea. I didn't need singletons for a long time.

Templates should work for singletons, because each instantiated template becomes a distinct type. Similar to monomorphization in Rust.

Reply:
I added a comment regarding the source of this code, and replaced Q_ASSERT with DEBUG_ASSERT.